enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,454 @@
+
+Phase 2 is more of an open ended process leveraging the new profile to
+optimize the configuration of the individual components for single
+node developer clusters. In general, each component will be addressed
+independently and possibly multiple times, first to eliminate the
+post-install patching by SNC, and then to enable additional
+optimizations. During implementation of this phase, it is expected
+that patterns will emerge when common approaches apply across
+different components.
+
+### Risks and Mitigations
+
+One risk is that all parts of the implementation of the cluster
+profiles
+[enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+do not get merged in a timely manner. Implementing the
+single-node-developer cluster profile requires the ability to specify
+a non-default cluster profile to the OpenShift installer, and for this
+to result in only the manifests annotated for the specified cluster
+profile to be processed by the resulting cluster. It's not clear at
+this time which team is signed up to implement the cluster profile
+support, but developer tools team members may be available to assist
+in writing code, and definitely will participate in the review
+process.
+
+Another risk is that it will take too long to get the initial
+annotations merged to all the needed OpenShift component repositories
+during the first phase of development. Until all needed annotations
+are merged, the SNC tooling cannot be updated to make use of the
+single-node-developer cluster profile, as needed components would be
+excluded from the cluster. The review process for this enhancement
+should help raise awareness across the various teams that will need to
+review and merge these patches.
+
+A final risk is that new components will be added to OpenShift after
+the completion of phase 1 that would be useful in single node
+developer clusters, but will not have the necessary annotation added
+due to lack of awareness or concern for the CRC product. One possible
+mitigation might be to alter the cluster profile mechanism to require
+that all manifests explicitly have either include or exclude
+annotations for each available profile.
+
+## Design Details
+
+### Open Questions
+
+N/A
+
+### Test Plan
+
+The first phase should not require any additional test cases, as the
+annotation for the single node developer cluster will be applied to
+existing manifest files that are already tested. Once these
+annotations are all merged and the SNC tooling is updated to have the
+OpenShift installer use the new cluster profile, the existing SNC
+tests and CRC product tests will verify that nothing has changed in
+the resulting cluster.
+
+The testing required during the second phase will depend on what
+changes are made to each individual component. If a specialized
+manifest is added for a component, an appropriate subset of existing
+tests for that component should ideally be run using the new
```

Comment:
Are you advocating this test runs per PR or as some periodic job?  Can you provide a pointer to what tests CRC runs today?

Reply:
In cases where specialized features or config knobs are added to a component specifically for this profile, I'd think those should be tested similarly to any other similar feature in the component. In previous projects I've worked on, that would involve at least basic UT coverage that runs on each PR, but I'm not yet very familiar with the OpenShift testing strategy, so could use some guidance here. 

I will investigate the current CRC testing strategy in more detail, and followup here with the requested info.

Reply:
Clarified component CI testing and added sections on conformance test job in update.